### PR TITLE
added basic sprite handling modules

### PR DESCRIPTION
**Pull request by pietrykovsky**
**Features**:
- created `SpriteObject` class representing 2d objects data
- added `ObjectManager` and `ObjectRenderer` classes for managing and rendering 2D sprites
- fixed sudden framerate drop when being very close to the rendered sprite or wall (scaling issue)

### CODE REVIEW

File: src/world_renderer.py [hunk 1]
```diff
@@ -20,7 +20,7 @@ def __init__(self, screen: pygame.Surface, raycaster: "Raycaster", map: "Map"):
         self.map = map
 
     def _calculate_shade(
-        self, color: tuple[int, int, int], distance: float
```
Comment (pietrykovsky) [inline src/world_renderer.py:23]: removed type hint (probably by mistake?), please add it back

File: src/world_renderer.py [hunk 2]
```diff
@@ -20,7 +20,7 @@ def __init__(self, screen: pygame.Surface, raycaster: "Raycaster", map: "Map"):
...
+        self, color: tuple[int, int, int], distance
```
Comment (pietrykovsky) [inline src/world_renderer.py:23]: removed type hint (probably by mistake?), please add it back

File: src/object_renderer.py [hunk 3]
```diff
@@ -49,11 +50,27 @@ def draw_object(self, obj: "SpriteObject"):
             + Settings().SCREEN_WIDTH // 2
             - width // 2
         )
-        screen_y = Settings().SCREEN_HEIGHT // 2 - height // 2
+        if height < Settings().SCREEN_HEIGHT:
+            height_shift = height * 0.27
```
Comment (pietrykovsky) [inline src/object_renderer.py:54]: where does the `0.27' value come from?

**Comment by pietrykovsky** [comment]
I spotted an issue with objects "jumping" when we are close to them

File: src/world_renderer.py [hunk 4]
```diff
@@ -76,44 +76,84 @@ def _draw_background(self):
                 1,
             )
 
+    def _draw_column(
```
Comment (pietrykovsky) [inline src/world_renderer.py:79]: this method has too many parameters, we should simplify it

File: src/world_renderer.py [hunk 5]
```diff
@@ -76,44 +76,84 @@ def _draw_background(self):
...
+        self.screen.blit(column, (col_x, y_pos))
+
+        shade_color = self._calculate_shade((255, 255, 255), ray_length)
+        shading_surface = pygame.Surface(column.get_size()).convert_alpha()
+        shading_surface.fill(shade_color + (128,))
+        self.screen.blit(
+            shading_surface, (col_x, y_pos), special_flags=pygame.BLEND_RGBA_MULT
+        )
+
+    def _prepare_column(
```
Comment (pietrykovsky) [inline src/world_renderer.py:100]: this method has too many parameters and we should simplify it

File: src/world_renderer.py [hunk 6]
```diff
@@ -76,44 +76,84 @@ def _draw_background(self):
...
+        self,
+        wall_texture: dict[int, pygame.Surface],
```
Comment (pietrykovsky) [inline src/world_renderer.py:102]: this type hint is wrong, we pass pygame.Surfance instance as an argument here

**Comment by pietrykovsky** [comment]
Maybe we could combine `WorldRenderer` and `ObjectRenderer`, create a _common collection_ of objects to draw (walls, sprite objects, etc.) and render them from farthest to closest? This way we could easily render the sprite objects behind the walls. The main challenge is to create a good column representation that would be stored in the collection.

**Comment by Milosz-cat** [comment]
good job

**Comment by Milosz-cat** [comment]
good job michal

**Pull request merged**